appinventor/components/src/com/google/appinventor/components/runtime/SpeechRecognizer.java

```diff
@@ -103,11 +126,67 @@ public void BeforeGettingText() {
   }
 
   /**
-   * Simple event to raise after the VoiceReco activity has returned
+   * Simple event to raise after the VoiceReco activity has returned.
+   * Partial is 'false' when SpeechRecognizer stops automatically after listening,
+   * else it is 'true' if SpeechRecognizer returns partial results.
    */
   @SimpleEvent
-  public void AfterGettingText(String result) {
-    EventDispatcher.dispatchEvent(this, "AfterGettingText", result);
+  public void AfterGettingText(String result, boolean partial) {
+    EventDispatcher.dispatchEvent(this, "AfterGettingText", result, partial);
+  }
+
+  /**
+   * Method from SpeechListener interface.
+   */
+  @Override
+  public void onPartialResult(String text) {
+    result = text;
+    AfterGettingText(result, true);
+  }
+
+  /**
+   * Method from SpeechListener interface.
+   */
+  @Override
+  public void onResult(String text) {
+    result = text;
+    AfterGettingText(result, false);
+    onDelete();
```

Comment:
Is it necessary to do this here? The semantics of onDelete are supposed to be such that it's only called when the component is being removed from the screen in the companion app.

Reply:
Okay, yes I get that. Actually, I thought onDelete() needs to be explicitly called, which, I guess, is not the case when I checked in other classes. But, if it is called automatically, when does that happens?